### DISCUSSION TITLE
Why is it that when I duplicate a block with attributes, the attributes are not preserved?

### DISCUSSION THREAD

**Discussion by Kzuy1** [mozman/ezdxf#892] [Q&A] [answered]
Hello Everyone,

When I try to insert/duplicate a block with attributes, it appears duplicated, but without its attributes.

![seu_arquivo_duplicado (2) dxf](https://github.com/mozman/ezdxf/assets/85893705/259adfc3-74b9-467c-8ce1-f3407102fdfe)

`import ezdxf
import openpyxl

doc = ezdxf.readfile("Template REDECAM-BLOCO-BRANCO.dxf")

bloco_original = doc.blocks.get('REDECAM-DISTINTA_monolingua')

msp = doc.modelspace()
if bloco_original:  # Adicionar várias referências do bloco no modelspace
  msp = doc.modelspace()
  for indice in range(5):
    pos = (0, 150 + 6 * indice, 0)
    msp.add_blockref(bloco_original.name, insert=pos)

  doc.saveas('seu_arquivo_duplicado.dxf')
else:
  print('O bloco original não foi encontrado')

doc.saveas("Drawing2_updated.dxf")
`

**Comment by mozman** [comment, marked as answer]
ATTRIB entities must be attached to a block reference:  https://ezdxf.mozman.at/docs/tutorials/blocks.html#
